operator/v1/types_ingress.go

```diff
@@ -513,6 +520,59 @@ type IngressControllerLogging struct {
 	Access *AccessLogging `json:"access,omitempty"`
 }
 
+// IngressControllerHTTPHeaderPolicy is a policy for setting HTTP headers.
+//
+// +kubebuilder:validation:Enum=Append;Replace;IfNone;Never
+type IngressControllerHTTPHeaderPolicy string
+
+const (
+	// AppendHTTPHeaderPolicy appends the header, preserving any existing header.
+	AppendHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Append"
+	// ReplaceHTTPHeaderPolicy sets the header, removing any existing header.
+	ReplaceHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Replace"
+	// IfNoneHTTPHeaderPolicy sets the header if it is not already set.
+	IfNoneHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "IfNone"
+	// NeverHTTPHeaderPolicy never sets the header, preserving any existing
+	// header.
+	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
+)
+
+// IngressControllerHTTPForwardedHeaderPolicy describes how the HTTP Forwarded
+// header and related headers are handled.
+type IngressControllerHTTPForwardedHeaderPolicy struct {
+	// policy specifies when and how the IngressController sets the
+	// Forwarded, X-Forwarded-For, X-Forwarded-Host, X-Forwarded-Port,
+	// X-Forwarded-Proto, and X-Forwarded-Proto-Version headers.  The value
+	// may be one of the following:
+	//
+	// * "Append", which specifies that the IngressController appends the
+	//   headers, preserving existing headers.
+	//
+	// * "Replace", which specifies that the IngressController sets the
+	//   headers, replacing any existing Forwarded or X-Forwarded-* headers.
+	//
+	// * "IfNone", which specifies that the IngressController sets the
+	//   headers if they are not already set.
+	//
+	// * "Never", which specifies that the IngressController never sets the
+	//   headers, preserving any existing headers.
+	//
+	// By default, the policy is "Append".
+	//
+	// +optional
+	Policy IngressControllerHTTPHeaderPolicy `json:"policy,omitempty"`
+}
+
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
+	// forwarded describes how the Forwarded HTTP header and related headers
+	// are handled.
+	//
+	// +optional
```

Comment:
optional fields need to describe what happens on empty.  Empty is a straight pass through or pass through while handling auth headers or what?